cypress/support/pages/product.pageObject.js

```diff
@@ -0,0 +1,17 @@
+/// <reference types="Cypress" />
+
+import PageObject from '../PageObject';
+
+class ProductPageObject extends PageObject {
```

Comment:
Looks like your `ProductPageObject` class contains methods for the home page, and for the product page at the same time. Page Object Model must describe a single page or a single component using own methods.
Avoid mixing methods from different entities.